src/client/httpClient.js

```diff
@@ -30,15 +30,15 @@ export default (
     });
   });
 
-  store.on('watch', (query, options) => {
+  store.on('watch', async (query, options) => {
```

Comment:
This is actually wrong. The watch handler must not be an async function (aka function returning a promise), it must be an async generator function (`async function*`) or a function that returns an async iterable.

This is likely the reason for the test failure.

Reply:
If you do this, replace the `return makeStream(...)` statements with `yield* makeStream(...)`. The alternative is to use a bunch of .then()'s and keep it a regular non-async function.

Reply:
Yeah. My bad for jumping the gun. I'll fix this and also add a test case for watch.

Reply:
Test case a bit complex (involves mocking EventSource). Will add it some other day.